sunpy/net/jsoc/tests/test_jsoc.py

```diff
@@ -53,6 +53,8 @@ def test_return_query_args(client):
                         a.jsoc.Segment('Bp') & a.jsoc.Segment('magnetogram'))
     # Because res.query_args is list that contains dict
     assert 'primekey' in res.query_args[0]
+    assert "magnetogram" in res.keys()
+    assert res["magnetogram"][0].startswith("/SUM")
```

Comment:
Shouldn't we check for the existence of 'BP' as well since it is also a segment? I also wonder if a more comprehensive test wouldn't be running a query with and without passing `a.jsoc.Segment` and checking that the length is different / the segment names are now columns with the expected values.

Reply:
Sounds good, I will do so.

Reply:
I created a new test with these changes and undid the changes for this test.